docs/non_native_field_arithmetic.md

```diff
@@ -0,0 +1,165 @@
+# Non-Native Field Arithmetic
+
+**References:**
+
+- [Aztec Emulated Field and Group Operations for Efficient Large Number Arithmetic](https://hackmd.io/@kullervo/cryptographyExplorations/%2F%40kullervo%2FbignumberCircuits)
+- [Non-Native Field Arithmetic](https://hackmd.io/@JkY-zACaSqerTtn_UwFjKg/SJZw6x75o)
+- [Field selection for recursive SNARKs](https://medium.com/delendum/field-selection-for-recursive-snarks-726ad56c3a3c)
+- [Foreign field arithmetic](https://www.youtube.com/watch?v=IBeo5epQoYo)
+
+## Motivation
+
+Sometimes, we want to represent modulo arithmetic with prime modulus $p$ over our prime finite field $\mathbb{F}_n$ . It's called
+non-native field arithmetic (or some other names like *foreign field, mismatched field, non-aligned field, wrong field*). Below are a
+few use cases:
+
+- **For ZK-Rollup after EIP-4844:** After [EIP-4844](eip_4844.md), this operation is required for proof systems that do not
+  use [BLS12-381](../terms/bls12-381.md) modulus to deal with blob transaction.
+- **Verify cryptographic constructs:** at the protocol level, it is common to verify cryptographic constructs (hashes, signatures,
+  proofs) that may be defined over an arbitrary field ($\mathbb{F}_2, \mathbb{F}_q, ...$) inside our field $\mathbb{F}_n$.
+- **Proof aggregation:** merge multiple proofs and create a single proof from them to save resource.
+
+In this article, we will take advantage of [Chinese Remainder Theorem](../terms/chinese_remainder_theorem.md) (CRT)
+and [Plookup](reed_solomon_fingerprinting.md) to solve the problem, separately.
+
+## Problem
+
+We are interested in building a circuit for $a \cdot b \pmod p$ over our field $\mathbb{F}_n$ (the same for $a+b$ with our trick).
+Typically, a witness $(a,b,r,q,p)$ is produced and the integer constraint $a \cdot b - p \cdot q - r = 0$ is included as part of
+the [polynomial commitment](../terms/polynomial-commitment/000_polynomial_commitment.md).
+
+If we are willing to deal with values in the range $\lbrack 0, p^2)$ in our circuit constraints, we can simply ensure that $a \cdot b -
```

Comment:
```suggestion
We need to avoid the computation of $a \cdot b$ or $p \cdot
 q$, because it is inefficient to compute them if $p$ is a large prime number
```